bufconn.go

```diff
@@ -34,17 +34,11 @@ func Wrap(wrapped net.Conn) Conn {
 
 func (c *conn) Read(b []byte) (n int, err error) {
 	if c.br != nil {
-		br := c.br
-		n = c.br.Buffered()
-		if n > len(b) {
-			n = len(b)
-			b = b[:n]
-		} else {
+		n, err = c.br.Read(b)
```

Comment:
Slight simplification. `bufio.Reader.Read()` returns whatever in the buffer until there's none left. Even if the implementation changes later, the logic here still doesn't break things.

Reply:
The problem is that without my logic, we'll keep reading via bufio.Reader which will keep filling its buffer. That's less efficient than reading directly from the connection, since it involves a copy. I'll make a comment about that in the code.

Reply:
Oh, interesting, I looked at the implementation and you're right, bufio's `Read` method doens't actually fill the buffer! Good simplification.